### PR TITLE
CANDLEPIN-150: Add pagination to getHypervisors endpoint

### PR DESCRIPTION
**Pull request by Januson**
- Added paging params to getHypervisors endpoint spec
- Added pagination for getting hypervisors by owner
- Added pagination for getting hypervisors by ids
- Removed unused code from consumer curator

### CODE REVIEW

**Review by joshmalbrecht — CHANGES_REQUESTED** [review]
Just noticed a change to use the ConfigProperties.MAX_PAGING_SIZE configuration

File: src/main/java/org/candlepin/resource/OwnerResource.java [hunk 1]
```diff
@@ -167,6 +166,7 @@ public class OwnerResource implements OwnerApi {
     private static final Logger log = LoggerFactory.getLogger(OwnerResource.class);
 
     private static final Pattern AK_CHAR_FILTER = Pattern.compile("^[a-zA-Z0-9_-]+$");
+    private static final int MAX_RESULTS = 1000;
```
[the remark this answers is not in the document]
Comment (Januson) [reply src/main/java/org/candlepin/resource/OwnerResource.java]: Done.

File: src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 2]
```diff
@@ -1151,55 +946,133 @@ public List<String> getInactiveConsumerIds(Instant lastCheckedInRetention, Insta
...
+            .getSingleResult().intValue();
+    }
+
+    private static List<String> toLowerCase(Collection<String> hypervisorIds) {
+        List<String> lowerCaseHypervisorIds = hypervisorIds.stream()
+            .distinct()
+            .map(String::toLowerCase)
+            .sorted()
+            .toList();
+        return lowerCaseHypervisorIds;
```
Comment (joshmalbrecht) [inline src/main/java/org/candlepin/model/ConsumerCurator.java:1025]: No need to for the `lowerCaseHypervisorIds` variable, we can just return the list:

```
    private static List<String> toLowerCase(Collection<String> hypervisorIds) {
        return hypervisorIds.stream()
            .distinct()
            .map(String::toLowerCase)
            .sorted()
            .toList();
    }
```

**Review by joshmalbrecht — CHANGES_REQUESTED** [review]
Just one small change to remove a temporary variable

**Review by joshmalbrecht — APPROVED** [review]

**Pull request merged**